lib/rules/dom-elements-style-is-object.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * @fileoverview Enforce style prop value being an object and havng sting values
+ * @author David Petersen
+ */
+'use strict';
+
+var variableUtil = require('../util/variable');
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Enforce style prop value being an object and havng sting values',
```

Comment:
It would be nice for this rule to stick to the type of check that is made by React, which simply verifies that it is null or an object.

https://github.com/facebook/react/blob/85dcbf83/src/renderers/dom/shared/ReactDOMComponent.js#L204

For example, these should all be valid:

- Integer value, e.g. `style={{ width: 100 }}`
- Some string values and some integer values, e.g. `style={{ color: 'red', width: 100 }}`
- Interpolation in a value, e.g. `style={{ border: `1px solid ${primaryColor}` }}`
- Function call in the value, e.g. `style={{ border: borderFactory(1) }}`
- Variable as the value, e.g. `style={{ color: red }}` and `style={{ color }}`
- String concatenation in the value, e.g. `style={{ border: '1px solid ' + primaryColor }}`